pkg/store/kotsstore/version_store.go

```diff
@@ -787,43 +787,15 @@ func (s *KOTSStore) GetAppVersion(appID string, sequence int64) (*versiontypes.A
 	return v, nil
 }
 
-func (s *KOTSStore) GetAppVersions(appID string) ([]*versiontypes.AppVersion, error) {
-	db := persistence.MustGetDBSession()
-	query := `select app_id, sequence, update_cursor, channel_id, version_label, created_at, status, applied_at, kots_installation_spec, kots_app_spec, kots_license from app_version where app_id = $1`
-
-	rows, err := db.Query(query, appID)
+// GetLatestAppSequence returns the sequence of the latest app version.
+// This function handles both semantic and non-semantic versions.
+// If downloadedOnly param is set to true, the sequence of the latest downloaded app version will be returned.
+func (s *KOTSStore) GetLatestAppSequence(appID string, downloadedOnly bool) (int64, error) {
+	versions, err := s.FindDownstreamVersions(appID, downloadedOnly)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to query db")
+		return 0, errors.Wrap(err, "failed to get latest downstream version")
 	}
-
-	versions := []*versiontypes.AppVersion{}
-	for rows.Next() {
-		v, err := s.appVersionFromRow(rows)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to get app version from row")
-		}
-		versions = append(versions, v)
-	}
-
-	if err := rows.Err(); err != nil {
-		if err == sql.ErrNoRows {
-			return versions, nil
-		}
-		return nil, errors.Wrap(err, "failed to iterate over rows")
-	}
-
-	return versions, nil
-}
-
-func (s *KOTSStore) GetLatestAppVersion(appID string, downloadedOnly bool) (*versiontypes.AppVersion, error) {
-	downstreamVersions, err := s.FindDownstreamVersions(appID, downloadedOnly)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to find app downstream versions")
-	}
-	if len(downstreamVersions.AllVersions) == 0 {
-		return nil, errors.New("no app versions found")
-	}
-	return s.GetAppVersion(appID, downstreamVersions.AllVersions[0].ParentSequence)
+	return versions.AllVersions[0].ParentSequence, nil
```

Comment:
this should check that there is at least one version.

Reply:
the function that retrieves the versions already does